### PR TITLE
Add the translator nudge for non-Mag-16 locales

### PR DESCRIPTION
**Pull request by d-alleyne**
Related to Automattic/i18n-issues#850

## Proposed Changes

Adds a translation nudge on the Calypso home screen, and to the language switcher (in next commit)

## Why are these changes being made?

We removed the nudge on https://github.com/Automattic/wp-calypso/pull/91826, but this re-adds it so that it is more targeted 

## Testing Instructions

- Login into Calypso, and change the language to one whose translation percentage is less than 85%.
- Verify that the nudge is shown to the users

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/components/translator-invite/index.jsx [hunk 1]
```diff
@@ -28,50 +28,110 @@ export class TranslatorInvite extends Component {
 	recordClick = () =>
 		this.props.recordTracksEvent( 'calypso_translator_invitation', {
 			language: this.props.localizedLanguageNames[ this.props.locale ].en,
+			location: this.props.path,
 		} );
 
+	renderIcon() {
```
Comment (yuliyan) [inline client/components/translator-invite/index.jsx]: This method is currently unused.

File: client/components/translator-invite/index.jsx [hunk 2]
```diff
@@ -28,50 +28,110 @@ export class TranslatorInvite extends Component {
...
+		return <MaterialIcon icon="emoji_language" />;
+	}
+
+	renderHeader() {
```
Comment (yuliyan) [inline client/components/translator-invite/index.jsx]: This method is currently unused.

File: client/components/translator-invite/index.jsx [hunk 3]
```diff
@@ -28,50 +28,110 @@ export class TranslatorInvite extends Component {
...
+		return (
+			<p className="card-heading card-heading-20">{ translate( 'Translate WordPress.com' ) }</p>
+		);
+	}
+
+	renderLearnMoreLink = ( isInline = false ) => {
+		const { locale, translate } = this.props;
+		const link = (
+			<a
+				className="language-picker__modal-incomplete-locale-notice-learn-more"
```
Comment (yuliyan) [inline client/components/translator-invite/index.jsx]: According to the [BEM naming convention](https://github.com/Automattic/wp-calypso/blob/ce1a376af4bcc8987855ced4c961efacda6e1d32/packages/eslint-plugin-wpcalypso/docs/rules/jsx-classname-namespace.md?plain=1#L3), the class name here suggest that the link is an element of `language-picker`, whereas it's actually an element of `translator-invite`.

File: client/components/translator-invite/index.jsx [hunk 4]
```diff
@@ -28,50 +28,110 @@ export class TranslatorInvite extends Component {
...
 			);
+		} else if ( path === '/language-switcher' ) {
+			noticeText = translate( 'You can help translate WordPress.com into your language.' );
+		} else {
+			return null;
 		}
 
-		return null;
+		return (
+			<div className="language-picker__modal-incomplete-locale-notice-info">{ noticeText }</div>
```
Comment (yuliyan) [inline client/components/translator-invite/index.jsx]: According to the [BEM naming convention](https://github.com/Automattic/wp-calypso/blob/ce1a376af4bcc8987855ced4c961efacda6e1d32/packages/eslint-plugin-wpcalypso/docs/rules/jsx-classname-namespace.md?plain=1#L3), the class name here suggest that the notice text is an element of `language-picker`, whereas it's actually an element of `translator-invite`.

File: client/my-sites/customer-home/cards/features/app-promo/index.tsx [hunk 5]
```diff
@@ -1,15 +1,21 @@
...
+		<>
+			<AppPromo
+				title={ translate( 'Get our mobile app' ) }
+				className="customer-home__card"
+				subheader={ translate(
+					'Everything you need to build and grow your site from any device.'
+				) }
+				campaign="calypso-customer-home"
+			/>
+			<TranslatorInvite path="/home" />
```
Comment (yuliyan) [inline client/my-sites/customer-home/cards/features/app-promo/index.tsx]: Is there a reason to render the `TranslatorInvite` as part of the `AppPromoHome` card instead of rendering it as its own card?

**Review by yuliyan — COMMENTED** [review]
The translator nudge messages intended for the site home screen and the language picker, seem to have more differences than similarities. Using a single component to fit both use cases seems to have resulted in adding unnecessary complexity to a component that's supposed to simply render an icon with some text and a link.

I don't really see a major benefit of using a single component, so I think using two separate components would allow to drop some of the logic and might make the implementation a little easier to follow.

----

Apart from that, I've noticed some other problems:

* Some unit tests are failing.
* There are discrepancies with [the proposed the design](https://github.com/Automattic/i18n-issues/issues/850#issuecomment-2231355732) with regards to the margins, font sizes, and colors.
* The `TranslatorInvite` is currently not being used on the language picker, despite having the implementation for it.

File: client/components/translator-invite/index.jsx [hunk 6]
```diff
@@ -28,50 +28,110 @@ export class TranslatorInvite extends Component {
...
+
+	renderIconAndHeader = () => {
+		const { path, translate } = this.props;
+
+		switch ( path ) {
+			case '/home':
+				return (
+					<>
+						<MaterialIcon icon="emoji_language" />
+						<p className="card-heading card-heading-20">
```
Comment (yuliyan) [inline client/components/translator-invite/index.jsx]: Using the class names from [the CardHeading component](https://github.com/Automattic/wp-calypso/blob/trunk/client/components/card-heading/index.jsx#L12) will only provide the expected styling if the `CardHeading` is used elsewhere in the current client session so that it's stylesheet gets loaded, which may not always be the case.

File: client/components/translator-invite/index.jsx (hunk 1 above)
Comment (d-alleyne) [reply client/components/translator-invite/index.jsx]: Thanks for taking the time to review. 
A local commit had removed the unused methods. Implementing the Language switcher version is in progress.
> There are discrepancies with https://github.com/Automattic/i18n-issues/issues/850#issuecomment-2231355732 with regards to the margins, font sizes, and colors.

Good eye. I was mainly verifying that the layout and relative sizes were correct.

> I don't really see a major benefit of using a single component, so I think using two separate components would allow to drop some of the logic and might make the implementation a little easier to follow.

Will do.

File: client/components/translator-invite/index.jsx (hunk 4 above)
Comment (d-alleyne) [reply client/components/translator-invite/index.jsx]: Aah, I was unaware that we were following a specific convention. I was reusing [the classes](https://github.com/Automattic/wp-calypso/blob/trunk/client/components/language-picker/modal.tsx#L80) found in the Language Switcher when developing this component.

File: client/my-sites/customer-home/cards/features/app-promo/index.tsx (hunk 5 above)
Comment (d-alleyne) [reply client/my-sites/customer-home/cards/features/app-promo/index.tsx]: Initially, that was the plan. However, making a new card [appear on the Calypso Home page](https://github.com/Automattic/wp-calypso/blob/1384afc6779ff93ac95d9ebc443d93a9b6579b6d/client/my-sites/customer-home/locations/tertiary/manage-site/index.jsx#L4) was more complex than I expected. It was simpler to add the component to the AppPromoHome.

File: client/components/language-picker/modal.scss [hunk 7]
```diff
@@ -67,6 +67,19 @@
 	}
 }
 
+.language-picker__modal-incomplete-locale-nudge-text {
+	font-weight: normal;
+
+	.material-icon {
+		margin: 0 2px -4px 0;
+		transform: scale(0.8);
```
Comment (yuliyan) [inline client/components/language-picker/modal.scss]: The [<MaterialIcon /> component accepts `size` prop](https://github.com/Automattic/wp-calypso/blob/d2f640d00a8f4b4782dca1b9153454ed3fc30c43/packages/components/src/material-icon/index.tsx#L13). It's preferred to use that instead of scaling down the icon via `transform`.

File: client/components/language-picker/modal.scss [hunk 8]
```diff
@@ -67,6 +67,19 @@
...
+	}
+}
+
+.language-picker__modal-incomplete-locale-nudge-link {
+	font-weight: normal;
```
Comment (yuliyan) [inline client/components/language-picker/modal.scss]: Not: The `normal` seems to be inherited from the default styles, so it seems redundant to set it explicitly here.

File: client/components/translator-invite/style.scss [hunk 9]
```diff
@@ -1,20 +1,26 @@
...
 	max-width: 75%;
-	text-align: center;
-}
+	text-align: left;
 
-.is-section-signup .translator-invite__content,
-.is-section-signup .translator-invite__content a {
-	color: var(--color-text-inverted);
-}
+	&.notice-header {
```
Comment (yuliyan) [inline client/components/translator-invite/style.scss]: Since we've [previously mentioned](https://github.com/Automattic/wp-calypso/pull/92772#discussion_r1691070010) the [BEM naming convention](https://en.bem.info/methodology/naming-convention/), here's an example of the class names that I would go with to be compliant with it:

```css
.translator-invite__content { ... }
.translator-invite__heading { ... }
.translator-invite__subheading { ... }
.translator-invite__link { ... }
.translator-invite__icon { ... }
```

File: client/my-sites/customer-home/cards/features/app-promo/index.tsx [hunk 10]
```diff
@@ -4,12 +4,16 @@ import AppPromo from 'calypso/blocks/app-promo';
 const AppPromoHome = () => {
 	const translate = useTranslate();
 	return (
-		<AppPromo
-			title={ translate( 'Get our mobile app' ) }
-			className="customer-home__card"
-			subheader={ translate( 'Everything you need to build and grow your site from any device.' ) }
-			campaign="calypso-customer-home"
-		/>
+		<>
```
Comment (yuliyan) [inline client/my-sites/customer-home/cards/features/app-promo/index.tsx]: I believe the fragment here is not required. The formatting changes also seem unrelated to this PR.

File: client/my-sites/customer-home/locations/tertiary/manage-site/index.jsx [hunk 11]
```diff
@@ -63,6 +64,7 @@ const ManageSite = () => {
 						key: index,
 					} )
 			) }
+			<TranslatorInvite path="/home" />
```
Comment (yuliyan) [inline client/my-sites/customer-home/locations/tertiary/manage-site/index.jsx]: Would it make sense to move one level up, directly inside [the Home `<Tertiary />` location](https://github.com/Automattic/wp-calypso/blob/a56204b117473f4379ac0e8039adc2aea95e7406/client/my-sites/customer-home/locations/tertiary/index.jsx#L23)?

**Review by yuliyan — CHANGES_REQUESTED** [review]
### Visual remarks

#### Home screen

vxEFZmY0n331EBRA

1. The icon color is different from the design.
2. The margin between the two cards doesn't match.
3. Font-weight is different between the two cards. The link color is different from the design, but matches other links on the screen so I'm not sure which one should be used. The link underline has more offset in the design and the card above.
4. Missing padding on mobile viewport:
OPnBNkPD6lnRK7aV

#### Language picker

sidDlNp4RCaUPmrV

1. The icon is not aligned with the checkbox above it.
2. The color of the icon doesn't match the design.
3. There's less space between the percentage notice and the translator nudge.
4. I don't think it's ok to have the notice inside a label element: 
QlVFlS7hkbO2Yagy



### Other

* Unit tests are failing. Could you fix them?
* Since the components visibility is based on the translated percentage, it might be worth to follow up updating the languages data, as it seems like the [languages packages hasn't been updated in a while](https://github.com/Automattic/wp-calypso/blob/trunk/packages/languages/src/languages-meta.json).

File: client/components/language-picker/modal.scss (hunk 7 above)
Comment (d-alleyne) [reply client/components/language-picker/modal.scss]: The `size` prop depends on the `.svg` files of that [exact name](https://github.com/Automattic/wp-calypso/blob/e5b49b40406c4a8adc844e3fbab8c57d13e9e462/packages/components/src/material-icon/index.tsx#L20) existing [material-design-icons](https://github.com/Automattic/wp-calypso/blob/trunk/packages/material-design-icons/README.md) directory.

Comment (yuliyan) [reply client/components/language-picker/modal.scss]: Ah, I didn't see that. In this case, the `width` and `height` could be set in the CSS.

File: client/components/language-picker/modal.scss [hunk 12]
```diff
@@ -67,6 +67,17 @@
 	}
 }
 
+.language-picker__modal-incomplete-locale-nudge-text {
+	font-weight: normal;
+	/* stylelint-disable-next-line scales/font-sizes */
+	font-size: 0.9375rem; //typography-exception
```
Comment (yuliyan) [inline client/components/language-picker/modal.scss]: The `language-picker__modal-incomplete-locale-text` element above has a smaller font size of `0.875rem`. Let's use the same one here for consistency. It would also make the disabling the styleling rule obsolete.

File: client/components/language-picker/modal.tsx [hunk 13]
```diff
@@ -144,6 +135,21 @@ const LanguagePickerModal: React.FC< Props > = ( {
 							</span>
 						</FormLabel>
 					) }
+					{ selectedLanguage.isTranslatedIncompletely && (
```
Comment (yuliyan) [inline client/components/language-picker/modal.tsx]: The `selectedLanguage.isTranslatedIncompletely` seems redundant as it's already [one of the condition to render the parent element](https://github.com/Automattic/wp-calypso/pull/92772/files#diff-e2c927e8e979f0f2d70ee47e9b25071d8e3232a3c9a1842993925b8dfb5c8c11L125).

File: client/components/translator-invite/index.jsx [hunk 14]
```diff
@@ -28,50 +29,63 @@ export class TranslatorInvite extends Component {
...
+						<MaterialIcon className="translator-invite__icon" icon="emoji_language" />
+
+						<h2 className="translator-invite__heading">
+							{ translate( 'Translate WordPress.com' ) }
+						</h2>
+
+						{ this.renderNoticeLabelText() }
+
+						<a
+							className="translator-invite__link is-link"
```
Comment (yuliyan) [inline client/components/translator-invite/index.jsx]: The `is-link` class name doesn't seem to provide any styling, so I think it's not needed.

File: client/components/translator-invite/index.jsx [hunk 15]
```diff
@@ -28,50 +29,63 @@ export class TranslatorInvite extends Component {
...
 			return null;
 		}
 
+		const currentLanguage = languages.find( ( language ) => language.langSlug === locale );
+		const isTranslatedIncompletely = currentLanguage?.isTranslatedIncompletely;
+
 		return (
 			<div className="translator-invite">
-				{ this.renderNoticeLabelText() }
+				{ isTranslatedIncompletely && (
```
Comment (yuliyan) [inline client/components/translator-invite/index.jsx]: The component will render the empty `translator-invite` div and the `<QueryLanguageNames />` for languages with complete languages, which I suppose is not intended.

File: client/components/translator-invite/style.scss [hunk 16]
```diff
@@ -1,20 +1,32 @@
...
+	.translator-invite {
+		padding-inline-start: 20px;
+	}
+}
+
 .translator-invite__content {
-	margin: 15px auto;
-	font-size: $font-body-extra-small;
-	font-style: italic;
 	max-width: 75%;
```
Comment (yuliyan) [inline client/components/translator-invite/style.scss]: Is the max width property left out intentionally?

File: client/components/translator-invite/style.scss [hunk 17]
```diff
@@ -1,20 +1,32 @@
...
-	color: var(--color-text-inverted);
+.translator-invite__heading {
+	/* stylelint-disable-next-line scales/font-sizes */
+	font-size: 1.125rem; //typography-exception
+	font-weight: 500;
+	margin-block-end: 8px;
 }
 
-.is-section-signup .translator-invite__content a {
+.translator-invite__link {
```
Comment (yuliyan) [inline client/components/translator-invite/style.scss:24]: The font size should be `0.9375rem` and font weight `600` in order to match the styling in the app promo card above.

File: packages/material-design-icons/src/communication/outline-emoji_language-16px.svg
Comment (yuliyan) [inline packages/material-design-icons/src/communication/outline-emoji_language-16px.svg:1]: Is there any other difference than the size? If not, then the original icon can be resized with CSS where needed instead of introducing a separate SVG.

File: client/components/language-picker/modal.tsx (hunk 13 above)
Comment (d-alleyne) [reply client/components/language-picker/modal.tsx]: Good catch!

File: client/components/translator-invite/index.jsx (hunk 15 above)
Comment (d-alleyne) [reply client/components/translator-invite/index.jsx]: Actually, this was done to preserve existing behaviour, and fix [this test](https://github.com/Automattic/wp-calypso/blob/8464223a66e00048511ce73b27f39e02093a1742/client/components/translator-invite/test/index.js#L37).

Comment (yuliyan) [reply client/components/translator-invite/index.jsx]: I don't really see the point to render an empty DOM element in this case. Also, requesting the language names data that is not going to be used because the rendering condition is not fulfilled seems pointless too.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16458391

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @d-alleyne, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `%(languageName)s is only %(percentTranslated)d%% translated. Help translate WordPress into your language.`
- `Translate WordPress.com`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16458391).
